### ISSUE TITLE
Nullish or Falsey?

### ISSUE THREAD

**Issue by keithamus** [keithamus/proposal-array-compact#2]
This issue is to settle on whether the method should discard nullish (`null`, `undefined`) or falsey (`false`, `0`, `''`, `null`, `undefined`) values.

**Comment by keithamus** [comment]
There is an option to add both; i.e. adding `compactFalsey()` and `compactNullish`. Perhaps this is better as `discardFalsey()`, `discardNullish()`. See #1 for bikeshedding names.

**Comment by JamesMGreene** [comment]
I would agree that seeing examples of the following variations are also fairly common:
- `.filter(i => i != null)` / `.filter(i => i !== null)` ([~7K results](https://cs.github.com/?scopeName=All+repos&scope=&q=%2F.filter%5C%28%5Ba-z%5D%5Cs*%3D%3E%5Cs*%5Ba-z%5D%5Cs*%21%3D%3D%3F%5Cs*null%5C%29%2F+%28language%3AJavaScript+OR+language%3ATypeScript%29))
- `.filter(i => i != undefined)` / `.filter(i => i !== undefined)` ([~7K results](https://cs.github.com/?scopeName=All+repos&scope=&q=%2F.filter%5C%28%5Ba-z%5D%5Cs*%3D%3E%5Cs*%5Ba-z%5D%5Cs*%21%3D%3D%3F%5Cs*undefined%5C%29%2F+%28language%3AJavaScript+OR+language%3ATypeScript%29))

However, as you noted, those are nullish checks and _different_ from falsy checks.

**Comment by ljharb** [comment]
what would be interesting to note (but harder to determine) is how many of the falsy checks are in fact nullish checks in disguise.
